lib/tbd/psi.rb

```diff
@@ -1666,27 +1744,30 @@ def process(model = nil, argh = {})
         #   1x deratable ceiling
         edge[:surfaces].keys.each do |i|
           break     if is[:parapet]
+          break     if is[:roof   ]
           break unless deratables.size == 2
           break unless ceilings.key?(id)
           next      if i == id
           next  unless deratables.include?(i)
           next  unless walls.key?(i)
 
           s1      = edge[:surfaces][id]
-          s2      = edge[:surfaces][i]
+          s2      = edge[:surfaces][i ]
           concave = concave?(s1, s2)
           convex  = convex?(s1, s2)
           flat    = !concave && !convex
 
-          set[:parapet       ] = shorts[:val][:parapet       ] if flat
-          set[:parapetconcave] = shorts[:val][:parapetconcave] if concave
-          set[:parapetconvex ] = shorts[:val][:parapetconvex ] if convex
-           is[:parapet       ] = true
-
-          # set[:roof          ] = shorts[:val][:roof          ] if flat
-          # set[:roofconcave   ] = shorts[:val][:roofconcave   ] if concave
-          # set[:roofconvex    ] = shorts[:val][:roofconvex    ] if convex
-          #   is[:roof         ] = true
+          if argh[:parapet]
+            set[:parapet       ] = shorts[:val][:parapet       ] if flat
+            set[:parapetconcave] = shorts[:val][:parapetconcave] if concave
+            set[:parapetconvex ] = shorts[:val][:parapetconvex ] if convex
+             is[:parapet       ] = true
+          else
+            set[:roof       ] = shorts[:val][:roof       ] if flat
```

Comment:
TBD's argument hash (`argh`) always holds a **:parapet** key/value pair (`true` by default) for the whole model. This can be reset to `false` for the whole model (e.g. Measure boolean argument 'parapet'). This can be subsequently customized for:

- stories
- spacetypes
- spaces
- individual surfaces

... via TBD JSON/Hash specs.